tests/test_spider.py

```diff
@@ -113,6 +113,100 @@ def test_log(self):
             spider.log('test log msg', 'INFO')
         mock_logger.log.assert_called_once_with('INFO', 'test log msg')
 
+    def test_get_class_attributes_annotations(self):
+
+        class ParentTypedSpider(Spider):
+            param_int: int
+
+        class TypedSpider(ParentTypedSpider):
+            param_float: float
+
+        annotations = ParentTypedSpider.get_class_attributes_annotations()
```

Comment:
I'd expect `TypedSpider` to be used here (or in a separate test), otherwise I don't see how is the inheritance important in this test?